docs/ref/applications.txt

```diff
@@ -311,6 +311,11 @@ Methods
         separate from the production settings, ``manage.py test`` would still
         execute some queries against your **production** database!
 
+    .. versionchanged:: 5.0
+
+        Django will raise a ``RuntimeWarning`` when you interact with the
+        database before the app registry has been fully populated.
+
```

Comment:
Based on Carlton's comment in the ticket, ideally we'll explain or add a pointer about how the user can fix this warning.

Reply:
I've added a link to the troubleshooting section of the applications documentation to the warning message. I've also added a few paragraphs to the docs that reproduce the example from the [original ticket](https://code.djangoproject.com/ticket/33143), suggest using a `ModelChoiceField`, and point to Python's [warning filter](https://docs.python.org/3/library/warnings.html#the-warnings-filter) in the hopes that people could more easily discover the places in which they perform import-time queries.

Reply:
Removed the link to the troubleshooting section again because of @adamchainz's [comment](https://github.com/django/django/pull/16923#discussion_r1218613418) below.